File: 프로그래머스/1/42748. K번째수/K번째수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    for(vector<int> command : commands){
        int i = command[0] - 1;
        int j = command[1];
        int k = command[2] - 1;
        vector<int> tmp;
        for(int v = i; v < j; v++){
            tmp.push_back(array[v]);
        }
        sort(tmp.begin(), tmp.end());
        answer.push_back(tmp[k]);
    }
    
    return answer;
}

File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr, int[][] commands) {
        int n = arr.length;
        int m = commands.length;
        int[] ret = new int[m];
        for(int i = 0; i < m; i++){
            int[] cmd = commands[i];
            int a = cmd[0];
            int b = cmd[1];
            int c = cmd[2];
            a--;
            b--;
            c--;
            
            List<Integer> list = new LinkedList<>();
            for(int j = a; j <= b; j++){
                list.add(arr[j]);
            }
            Collections.sort(list);
            ret[i] = list.get(c);
        }
        return ret;
    }
}